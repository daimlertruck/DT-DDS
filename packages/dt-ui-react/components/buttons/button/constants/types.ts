import { ComponentSize } from '../../../../types/componentSize';

export type ButtonVariant = 'solid' | 'outlined' | 'text';

export type ButtonColor = 'primary' | 'error' | 'neutral';

export type ButtonSize = Extract<ComponentSize, 'medium' | 'large' | 'small'>;
