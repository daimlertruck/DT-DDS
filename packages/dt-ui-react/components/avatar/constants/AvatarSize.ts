export const AvatarSize = {
  Small: 'small',
  Medium: 'medium',
  Large: 'large',
} as const;

export type AvatarSize = (typeof AvatarSize)[keyof typeof AvatarSize];
