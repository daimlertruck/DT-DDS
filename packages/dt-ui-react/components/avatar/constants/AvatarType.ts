export enum AvatarType {
  Profile = 'profile',
  TeamMember = 'teamMember',
}
