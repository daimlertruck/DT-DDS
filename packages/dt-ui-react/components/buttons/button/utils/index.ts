export * from './buttonVariantColors';
export * from './buttonVariantStyles';
