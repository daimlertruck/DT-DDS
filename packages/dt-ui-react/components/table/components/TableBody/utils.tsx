import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
