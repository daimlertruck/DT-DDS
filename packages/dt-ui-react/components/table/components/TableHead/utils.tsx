import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
