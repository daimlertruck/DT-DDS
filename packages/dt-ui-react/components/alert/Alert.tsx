import { WarningIcon, InfoIcon, ErrorIcon } from '../../core/assets/svgs';
import { BaseProps } from '../../types';

import {
  AlertContentStyled,
  AlertIconStyled,
  AlertStyled,
} from './Alert.styled';
import { AlertType } from './constants';

export interface AlertProps extends BaseProps {
  type: AlertType;
}

const AlertIcons = {
  [AlertType.Error]: ErrorIcon,
  [AlertType.Warning]: WarningIcon,
  [AlertType.Info]: InfoIcon,
};

const Alert = ({ children, dataTestId, type, style }: AlertProps) => {
  const Icon = AlertIcons[type];
  const testId = dataTestId ?? 'alert';

  return (
    <AlertStyled data-testid={testId} type={type} style={style}>
      <AlertIconStyled type={type}>
        <Icon />
      </AlertIconStyled>
      <AlertContentStyled>{children}</AlertContentStyled>
    </AlertStyled>
  );
};

export default Alert;
