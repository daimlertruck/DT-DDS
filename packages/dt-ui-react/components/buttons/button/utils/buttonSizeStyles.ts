import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
