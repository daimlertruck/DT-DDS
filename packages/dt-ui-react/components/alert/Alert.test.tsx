import { render } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../utils';

import Alert from './Alert';
import { AlertType } from './constants';

describe('<Alert /> component', () => {
  const ProvidedAlert = withProviders(Alert);

  describe('when the alert type is warning', () => {
    it('should render a alert with some text and the right color and icon', () => {
      const { container } = render(
        <ProvidedAlert type={AlertType.Warning}>
          some warning message here
        </ProvidedAlert>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('when the alert type is error', () => {
    it('should render a alert with some text and the right color and icon', () => {
      const { container } = render(
        <ProvidedAlert type={AlertType.Error}>
          some error message here
        </ProvidedAlert>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('when the alert type is info', () => {
    it('should render a alert with some text and the right color and icon', () => {
      const { container } = render(
        <ProvidedAlert type={AlertType.Info}>
          some info message here
        </ProvidedAlert>
      );

      expect(container).toMatchSnapshot();
    });
  });
});
