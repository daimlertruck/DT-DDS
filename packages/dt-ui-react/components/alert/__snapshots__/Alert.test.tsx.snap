// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> component when the alert type is error should render a alert with some text and the right color and icon 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px 14px;
  gap: 18px;
  border: 2px solid #FF4A4A;
  border-radius: 2px;
}

.emotion-2 {
  width: 24px;
  height: 24px;
  color: #FF4A4A;
  mix-blend-mode: normal;
}

.emotion-4 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #151617;
  height: 100%;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="alert"
    type="error"
  >
    <div
      class="emotion-2 emotion-3"
      type="error"
    >
      <svg
        color="#FF4A4A"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.0711 19.0711C17.1957 20.9464 14.6522 22 12 22C9.34784 22 6.8043 20.9464 4.92893 19.0711C3.05357 17.1957 2 14.6522 2 12C2 9.34784 3.05357 6.8043 4.92893 4.92893C6.8043 3.05357 9.34784 2 12 2C14.6522 2 17.1957 3.05357 19.0711 4.92893C20.9464 6.8043 22 9.34784 22 12C22 14.6522 20.9464 17.1957 19.0711 19.0711Z"
          fill="currentColor"
        />
        <path
          d="M12.8797 13.6863L13.1157 6.48828H10.8997L11.0997 13.6863H12.8797ZM11.9907 17.5123C12.1621 17.5181 12.333 17.4894 12.4932 17.4281C12.6534 17.3667 12.7997 17.2739 12.9234 17.1551C13.0471 17.0362 13.1458 16.8938 13.2136 16.7362C13.2813 16.5786 13.3168 16.4091 13.318 16.2375C13.3191 16.066 13.2859 15.8959 13.2202 15.7375C13.1545 15.579 13.0577 15.4353 12.9356 15.3148C12.8135 15.1944 12.6684 15.0996 12.5091 15.0361C12.3497 14.9727 12.1792 14.9418 12.0077 14.9453C11.6733 14.9521 11.3547 15.0892 11.1198 15.3274C10.8849 15.5656 10.7522 15.886 10.75 16.2205C10.7478 16.555 10.8762 16.8772 11.1079 17.1184C11.3397 17.3597 11.6564 17.501 11.9907 17.5123Z"
          fill="white"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      some error message here
    </div>
  </div>
</div>
`;

exports[`<Alert /> component when the alert type is info should render a alert with some text and the right color and icon 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px 14px;
  gap: 18px;
  border: 2px solid #008DFC;
  border-radius: 2px;
}

.emotion-2 {
  width: 24px;
  height: 24px;
  color: #008DFC;
  mix-blend-mode: normal;
}

.emotion-4 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #151617;
  height: 100%;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="alert"
    type="info"
  >
    <div
      class="emotion-2 emotion-3"
      type="info"
    >
      <svg
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12C2 17.52 6.48 22 12 22C17.52 22 22 17.52 22 12C22 6.48 17.52 2 12 2ZM13 17H11V11H13V17ZM13 9H11V7H13V9Z"
          fill="white"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      some info message here
    </div>
  </div>
</div>
`;

exports[`<Alert /> component when the alert type is warning should render a alert with some text and the right color and icon 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px 14px;
  gap: 18px;
  border: 2px solid #FBD633;
  border-radius: 2px;
}

.emotion-2 {
  width: 24px;
  height: 24px;
  color: #FBD633;
  mix-blend-mode: normal;
}

.emotion-4 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  text-transform: none;
  color: #151617;
  height: 100%;
}

<div>
  <div
    class="emotion-0 emotion-1"
    data-testid="alert"
    type="warning"
  >
    <div
      class="emotion-2 emotion-3"
      type="warning"
    >
      <svg
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 6.49L19.53 19.5H4.47L12 6.49ZM12 2.5L1 21.5H23L12 2.5ZM13 16.5H11V18.5H13V16.5ZM13 10.5H11V14.5H13V10.5Z"
          fill="white"
        />
      </svg>
    </div>
    <div
      class="emotion-4 emotion-5"
    >
      some warning message here
    </div>
  </div>
</div>
`;
